pkg/auth/serviceaccount.go

```diff
@@ -94,3 +95,102 @@ func EnsureOIDCServiceAccountExistsForResource(ctx context.Context, serviceAccou
 
 	return nil
 }
+
+// EnsureOIDCServiceAccountRoleBindingExistsForResource
+// makes sure the given resource has an OIDC service account role binding with
+// an owner reference to the resource set.
+func EnsureOIDCServiceAccountRoleBindingExistsForResource(ctx context.Context, kubeclient kubernetes.Interface, gvk schema.GroupVersionKind, objectMeta metav1.ObjectMeta) error {
+	roleName := fmt.Sprintf("create-oidc-token")
+	roleBindingName := fmt.Sprintf("create-oidc-token", objectMeta.GetName())
+	roleBinding, err := kubeclient.RbacV1().RoleBindings(objectMeta.Namespace).Get(ctx, roleBindingName, metav1.GetOptions{})
+
+	// If the resource doesn't exist, we'll create it.
+	if apierrs.IsNotFound(err) {
+		logging.FromContext(ctx).Debugw("Creating OIDC service account role binding", zap.Error(err))
+
+		// Create the "create-oidc-token" role
+		CreateRoleForServiceAccount(ctx, kubeclient, gvk, objectMeta)
+
+		roleBinding := &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      roleBindingName,
+				Namespace: objectMeta.GetNamespace(),
+				Annotations: map[string]string{
+					"description": fmt.Sprintf("Role Binding for OIDC Authentication for %s %q", gvk.GroupKind().Kind, objectMeta.Name),
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "Role",
+				Name:     roleName,
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Namespace: objectMeta.GetNamespace(),
+					Name:      GetOIDCServiceAccountNameForResource(gvk, objectMeta),
+				},
+			},
+		}
```

Comment:
I would only place the logic to setup the instance of the role & roleBinding struct/object. The handling around creating/updating it as a kubernetes object should be done in a different module. Just check for example how it is done for the ApiServerSource deployment ("receive adapter"):

https://github.com/knative/eventing/blob/ef855c84d6117110430d6ac9c2817e22ddb35b0f/pkg/reconciler/apiserversource/resources/receive_adapter.go#L56

There you only define the object. Then the k8s object setup & reconciling is done in the reconciler: https://github.com/knative/eventing/blob/ef855c84d6117110430d6ac9c2817e22ddb35b0f/pkg/reconciler/apiserversource/apiserversource.go#L210